app/containers/ExplorePage.tsx

```diff
@@ -182,6 +135,11 @@ export default function ExplorePage() {
     return false;
   };
 
+  const getEventsCount = (array: Observation[]): number => {
+    const eventIds: number[] = array.map(o => o.event_id);
+    return eventIds.filter((v, i, a) => a.indexOf(v) === i && v !== undefined).length;
+  };
```

Comment:
1. This function doesn't need any data from outer scope - you can easily put it outside. Benefits: declutter the already huge `ExplorePage()`, make it testable.
2. This function takes runs `O(n^2)` and for large CSVs you can already feel the lag a bit. You can use the this snippet instead: `_(observations).map('event_id').without(undefined).uniq().size()` :slightly_smiling_face: